### PR TITLE
Calendar

### PR DESCRIPTION
**Pull request by burdick-alex**
added calendar feature

### CODE REVIEW

**Review by rhrussell — APPROVED** [review]
Just tried to open deployment on Heroku and all it shows is an error right from the beginning. Fix that then I can look at the other stuff.

**Review by rhrussell — CHANGES_REQUESTED** [review]
Just tried to open deployment on Heroku and all it shows is an error. Fix that then I can see the rest of your deployment. But your code on surface level looks fine, just need to fix that error.

File: app/javascript/packs/application.js [hunk 1]
```diff
@@ -9,6 +9,10 @@ require("@rails/activestorage").start()
 require("channels")
 
 
+
```
Comment (varvind) [inline app/javascript/packs/application.js:12]: Probably want to get rid of these extra line breaks

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -62,7 +62,10 @@
   <% end %>
 </nav>
   <div class='content'>
+    
     <%= yield %>
+    </div>
+
```
Comment (varvind) [inline app/views/layouts/application.html.erb:68]: get rid of extra line break

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -62,7 +62,10 @@
   <% end %>
 </nav>
   <div class='content'>
+
```
Comment (varvind) [inline app/views/layouts/application.html.erb:65]: same here

File: spec/api_spec.rb [hunk 4]
```diff
@@ -45,6 +47,7 @@
   end
   describe 'Get Specific Event' do
     it 'Valid' do
```
Comment (varvind) [inline spec/api_spec.rb:49]: maybe comment what this line means, seems kinda confusing at first glance

File: spec/api_spec.rb [hunk 5]
```diff
@@ -63,12 +66,14 @@
                                               })
     end
     it 'Invalid' do
+      travel_to Time.zone.local(2025, 1, 1, 0o1, 0o4, 44)
```
Comment (varvind) [inline spec/api_spec.rb:69]: same here

File: spec/calendar_spec.rb [hunk 6]
```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+RSpec.describe 'Create New Event', type: :system do
+  describe 'Create event' do
+    it 'test am event' do
+      travel_to Time.zone.local(2020, 3, 13, 0o1, 0o4, 44)
```
Comment (varvind) [inline spec/calendar_spec.rb:7]: same here

File: spec/calendar_spec.rb [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+      end
+
+      expect(page).to have_content('title1')
+
+      click_on 'Back to List'
+
+      click_on 'Check-in'
+    end
+    it 'test pm event' do
+      travel_to Time.zone.local(2020, 3, 13, 0o1, 0o4, 44)
```
Comment (varvind) [inline spec/calendar_spec.rb:54]: here too

File: spec/routes_spec.rb [hunk 8]
```diff
@@ -5,6 +5,7 @@
 RSpec.describe 'Event page', type: :system do
   describe 'Check Urls' do
     it 'View event with different url ids' do
+      travel_to Time.zone.local(2025, 1, 1, 0o1, 0o4, 44)
```
Comment (varvind) [inline spec/routes_spec.rb:8]: here

File: spec/routes_spec.rb [hunk 9]
```diff
@@ -23,6 +24,7 @@
       event2.destroy
     end
     it 'Visit event with different url ids' do
+      travel_to Time.zone.local(2025, 1, 1, 0o1, 0o4, 44)
```
Comment (varvind) [inline spec/routes_spec.rb:27]: here

File: spec/routes_spec.rb [hunk 10]
```diff
@@ -50,16 +52,18 @@
 RSpec.describe 'Participation Page', type: :system do
   describe 'Has Url' do
     it 'Checks the url parameters' do
+      travel_to Time.zone.local(2025, 1, 1, 0o1, 0o4, 44)
```
Comment (varvind) [inline spec/routes_spec.rb:55]: here

File: spec/routes_spec.rb [hunk 11]
```diff
@@ -78,6 +82,7 @@
   end
   describe 'Does not log in to Url' do
     it 'With other events password' do
+      travel_to Time.zone.local(2025, 1, 1, 0o1, 0o4, 44)
```
Comment (varvind) [inline spec/routes_spec.rb:85]: here

File: spec/tests_spec.rb [hunk 12]
```diff
@@ -5,6 +5,7 @@
 RSpec.describe 'Login page', type: :system do
   describe 'Login' do
     it 'Success' do
+      travel_to Time.zone.local(2025, 11, 1, 0o1, 0o4, 44)
```
Comment (varvind) [inline spec/tests_spec.rb:8]: here and the rest in the project :)

File: spec/tests_spec.rb [hunk 13]
```diff
@@ -25,30 +27,62 @@
...
       expect(page).to have_selector(:link_or_button, 'Details')
-      expect(page).to have_selector(:link_or_button, 'Event Check-in')
+      expect(page).to have_selector(:link_or_button, 'Check-in')
 
-      expect(page).to have_content('Event Title')
-      expect(page).to have_content('Event Place')
-      expect(page).to have_content('Event Description')
-      expect(page).to have_content('Start Time:')
-      expect(page).to have_content('End Time:')
+      # expect(page).to have_content('Event Title')
```
Comment (varvind) [inline spec/tests_spec.rb:66]: get rid of these commented bits

**Review by varvind — CHANGES_REQUESTED** [review]
Overall looks good, just some small things to make things neater

**Review by varvind — APPROVED** [review]
Everything looks good and all functionality tested on heroku

**Review by keithbryant3 — APPROVED** [review]
Everything seems to be good, though it would probably look a little better if you could get some padding at the bottom of the calendar so it isn't flush against the bottom of the page.

**Review by rhrussell — APPROVED** [review]
Everything looks good with the code and on Heroku. What I would suggest to do later on is to make it to where there is some space at the bottom between the calendar and the bottom of the page. That way the calendar is not right up against the bottom of the page. But other than that style critique, looks great. Good job!

**Pull request merged**